### PR TITLE
added stack class

### PR DESCRIPTION
**Pull request by marcelluethi**

### CODE REVIEW

File: Queue.java [hunk 1]
```diff
@@ -1,40 +1,40 @@
 class Queue {
```
Comment (colinfingerlin) [inline Queue.java:1]: No documentation provided.

Comment (colinfingerlin) [reply Queue.java:1]: As well as no commentary.

File: Queue.java [hunk 2]
```diff
@@ -1,40 +1,40 @@
...
-
-    void put(int x) {
-        if ((tail + 1) % size == head) {
-            overflow = true;
-        } else {
-            overflow = true;
-        }
-
-        if (overflow == true) {
-          q[tail]  =x;
```
Comment (colinfingerlin) [inline Queue.java:24]: Not consistent with code standards and not consistently formatted.

File: Queue.java [hunk 3]
```diff
@@ -1,40 +1,40 @@
...
+		boolean overflow = false;
+
+		Queue(int size) {
+				this.size = size;
+				q = new int[size];
+		}
+
+		void put(int x) {
+				if ((tail + 1) % size == head) {
+						overflow = true;
```
Comment (colinhex) [inline Queue.java:18]: Overflow is set to true but has no opportunity to be set back to false.

File: Queue.java [hunk 4]
```diff
@@ -1,40 +1,40 @@
...
+				} else {
+						overflow = true;
+				}
+
+				if (overflow == true) {
+					q[tail]  =x;
```
Comment (colinhex) [inline Queue.java:24]: Code formatting is substandard.

File: Queue.java [hunk 5]
```diff
@@ -1,40 +1,40 @@
...
-            head = (head + 1)% size;
-            return x;
-        }
-    }
+		int q[];
+		int head =0;
+		int tail = 0;
+		int size;
+
+		boolean underflow = false;
```
Comment (colinhex) [inline Queue.java:8]: Maybe put them private.

**Pull request closed without merging**